cloudinit/config/cloud-init-schema.json

```diff
@@ -1343,6 +1343,305 @@
+              },
+              "additionalItems": false,
+              "minItems": 1,
+              "minProperties": 1,
+              "additionalProperties": {
+                "oneOf": [
+                  {"type": "string"},
+                  {"type": "array", "items": {"type": "string"}}
+                ]
+              }
+            },
+            "squashfuse_in_container": {
+              "type": "boolean",
+              "default": false,
+              "description": "DEPRECATED: This key is no longer needed and will be removed in a future version of cloud-init."
+            }
+          }
+        }
+      }
+    },
+    "cc_spacewalk": {
+      "type": "object",
+      "properties": {
+        "spacewalk": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "server": {
+              "type": "string",
+              "description": "The Spacewalk server to use"
+            },
+            "proxy": {
+              "type": "string",
+              "description": "The proxy to use when connecting to Spacewalk"
+            },
+            "activation_key": {
+              "type": "string",
+              "description": "The activation key to use when registering with Spacewalk"
+            }
+          }
+        }
+      }
+    },
+    "cc_ssh_authkey_fingerprints": {
+      "type": "object",
+      "properties": {
+        "no_ssh_fingerprints": {
+          "type": "boolean",
+          "default": false,
+          "description": "If true, SSH fingerprints will not be written. Default: ``false``"
+        },
+        "authkey_hash": {
+          "type": "string",
+          "default": "sha256",
+          "description": "The hash type to use when generating SSH fingerprints. Default: ``sha256``"
+        }
+      }
+    },
+    "cc_ssh_import_id": {
+      "type": "object",
+      "properties": {
+        "ssh_import_id": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "The SSH public key to import"
+          }
+        }
+      }
+    },
+    "cc_ssh": {
+      "type": "object",
+      "properties": {
+        "ssh_keys": {
+          "type": "object",
+          "description": "A dictionary entries for the public and private host keys of each desired key type. Entries in the ``ssh_keys`` config dict should have keys in the format ``<key type>_private``, ``<key type>_public``, and, optionally, ``<key type>_certificate``, e.g. ``rsa_private: <key>``, ``rsa_public: <key>``, and ``rsa_certificate: <key>``. Not all key types have to be specified, ones left unspecified will not be used. If this config option is used, then separate keys will not be automatically generated. In order to specify multiline private host keys and certificates, use yaml multiline syntax.",
+          "patternProperties": {
+            "^(dsa|ecdsa|ecdsa-sk|ed25519|ed25519-sk|rsa)_(public|private|certificate)$": {
```

Comment:
This schema looks ok,  but the corresponding code in [[cc_ssh.py will silently skip any key type that is not in GENERATE_KEY_NAMES](https://github.com/canonical/cloud-init/blob/main/cloudinit/config/cc_ssh.py#L205-L206)

I think we need to properly handle these other keys and integration testing of them to make sure all is well. Since it's ssh-related I'd like that work to be a separate PR to make sure we get good eyes on it separate from schema coverage

Reply:
I'll make a ticket